server/src/main/scala/drt/server/feeds/lhr/live/LHRLiveFeed.scala

```diff
@@ -0,0 +1,247 @@
+    }
+
+    SDate.tryParseString(fixedDate) match {
+      case Success(sd) => sd.toISOString()
+      case Failure(f: Throwable) => ""
+    }
+  }
+
+
+  case class LHRLiveArrival(
+                             FLIGHTNUMBER: String,
+                             TERMINAL: String,
+                             FLIGHTSTATUS: String,
+                             STAND: String,
+                             OPERATOR: String,
+                             AIRCRAFTTYPE: String,
+                             AIRPORTCODE: String,
+                             COUNTRYCODE: String,
+                             SCHEDULEDFLIGHTOPERATIONTIME: String,
+                             ESTIMATEDFLIGHTOPERATIONTIME: String,
+                             ESTIMATEDFLIGHTCHOXTIME: String,
+                             ACTUALFLIGHTCHOXTIME: String
+                           )
+
+  case class LHRFlightPax(
+                           FLIGHTNUMBER: String,
+                           SCHEDULEDFLIGHTOPERATIONTIME: String,
+                           MAXPASSENGERCOUNT: String,
+                           TOTALPASSENGERCOUNT: String,
+                           ACTUALDIRECTPASSENGERCOUNT: String,
+                           ACTUALTRANSFERPASSENGERCOUNT: String,
+                           ACTUALT2INTCOUNT: String,
+                           ACTUALT2DOMCOUNT: String,
+                           ACTUALT3INTCOUNT: String,
+                           ACTUALT3DOMCOUNT: String,
+                           ACTUALT4INTCOUNT: String,
+                           ACTUALT4DOMCOUNT: String,
+                           ACTUALT5INTCOUNT: String,
+                           ACTUALT5DOMCOUNT: String
+                         )
+
+  val statusCodesToDesc = Map(
+    "AB" -> "Airborne",
+    "BD" -> "Boarding",
+    "CX" -> "Cancelled",
+    "CL" -> "Flight Cleared",
+    "DV" -> "Diverted",
+    "ES" -> "Estimated",
+    "EX" -> "Expected",
+    "FB" -> "First bag",
+    "FS" -> "Final approach",
+    "GC" -> "Gate closed",
+    "GO" -> "Gate open",
+    "LB" -> "Last Bag",
+    "LC" -> "Last call",
+    "LD" -> "Landed",
+    "OC" -> "On Chocks",
+    "OV" -> "OV",
+    "NI" -> "Next Info",
+    "OS" -> "Overshoot",
+    "RS" -> "Return to stand",
+    "SH" -> "Scheduled",
+    "TX" -> "Taxied",
+    "ZN" -> "Zoning",
+    "**" -> "Deleted"
+  )
+
+  abstract case class LHRLiveFeedConsumer(apiUri: String, subscriptionKey: String, implicit val system: ActorSystem) extends WithSendAndReceive with LHRLiveFeedParserProtocol {
+
+    import system.dispatcher
+
+    implicit val timeout = Timeout(1 minute)
+
+    val logResponse: HttpResponse => HttpResponse = resp => {
+
+      log.info(s"Got a response from LHR Live API: $resp")
+      if (resp.status.isFailure) {
+        log.warn(s"Error when reading LHR Live API ${resp.headers}, ${resp.entity.data.asString}")
+      }
+
+      resp
+    }
+
+    val arrivalsPipeline: HttpRequest => Future[List[List[LHRLiveArrival]]] = (
```

Comment:
could this and paxPipeline be the same function using a generic to specify the type?